homeassistant/components/modem_callerid/sensor.py

```diff
@@ -28,33 +25,47 @@
     }
 )
 
+SERVICE_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.comp_entity_ids})
+
+_LOGGER = logging.getLogger(__name__)
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up modem caller ID sensor platform."""
 
-    name = config.get(CONF_NAME)
-    port = config.get(CONF_DEVICE)
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Modem Caller ID sensor."""
+    name = entry.data[CONF_NAME]
+    device = entry.data[CONF_DEVICE]
+    api = hass.data[DOMAIN][entry.entry_id][DATA_KEY_API]
+    sensor = [
+        ModemCalleridSensor(
+            hass,
+            api,
+            name,
+            device,
+            entry.entry_id,
+        )
+    ]
+    async_add_entities(sensor, True)
```

Comment:
```suggestion
    async_add_entities([
        ModemCalleridSensor(
            hass,
            api,
            name,
            device,
            entry.entry_id,
        )
    ])
```

I think the `, True` is not needed here since this doesn't poll